### ISSUE TITLE
auro-modal: remove enforced header

### ISSUE THREAD

**Issue by blackfalcon** [AlaskaAirlines/auro-dialog#18]
## Is your feature request related to a problem? Please describe.
When importing content from other resources, there is a conflict to the header of the dialog. In these cases, the header is actually in the content and the pre-formatted dialog header is not needed. 

## Describe the solution you'd like
If there is no header content in the slot, this area will collapse. Much in the same way that the footer does. 

### A dialog in the Auro site with the current support:
![Screen Shot 2020-11-05 at 1 16 05 PM](https://user-images.githubusercontent.com/181089/98297413-37047c00-1f69-11eb-836a-9b4e2b4abe52.png)

### With the suggested update
![Screen Shot 2020-11-05 at 1 18 01 PM](https://user-images.githubusercontent.com/181089/98297528-63b89380-1f69-11eb-94bd-706edab13feb.png)

**Comment by blackfalcon** [comment]
This update is requesting input from design prior to making updates.

**Comment by geoffrich** [comment]
How will the dialog label be set in this case?

Currently we set aria-labelledby to the id of the h1 w/ the slot (irrelevant attributes removed). If this is empty or removed, the dialog no longer has a label.

```html
 <dialog aria-labelledby="dialog-header">
  <div class="dialog-header">
    <h1 id="dialog-header">
      <slot name="header">Default header ...</slot>
    </h1>
  </div>
```

Options I see:
1. Programmatically determine (get the first heading element in the slotted content)
2. Have the user pass aria-label or aria-labelledby as a prop. Log a warning/error if heading not provided and label not provided.

**Comment by blackfalcon** [comment]
> Have the user pass aria-label or aria-labelledby as a prop. Log a warning/error if heading not provided and label not provided.

The issue there assumes that either the user has control over the content or there is an ID on the header node. In the use case I illustrated I have neither.

**Comment by blackfalcon** [comment]
Going with this, we probably need a more robust management of the dialog header. Meaning, that if this content node is not addressed, there should be an error. 

In addition to this, we can still populate this node, then the request moves from ignoring the content node to hiding the node from sight.

**Comment by geoffrich** [comment]
Yeah, we definitely need something for the label. So I like the idea of requiring header to be populated for a11y, while exposing the option to visually hide it.

Even if the consumer doesn't control the content, I would assume that they have some idea of what content is in the modal and can provide a decent enough label for it.
